Entity/JobLog.php

```diff
@@ -157,179 +135,112 @@ public function toCompressedArray()
         ];
     }
 
-    /**
-     * @return string
-     */
-    public function getCommand()
+    public function getCommand(): string
     {
         return $this->command;
     }
 
-    /**
-     * @return string
-     */
-    public function getTopic()
+    public function getTopic(): string
     {
         return $this->topic;
     }
 
-    /**
-     * @return string
-     */
-    public function getAdded()
+    public function getAdded(): ?\DateTime
     {
         return $this->added;
     }
 
-    /**
-     * @return string
-     */
-    public function getCompleted()
+    public function getCompleted(): ?\DateTime
     {
         return $this->completed;
     }
 
     /**
-     * The duration of the job in seconds
-     *
-     * @return integer
+     * Returns duration in seconds
+     * @return int
      */
-    public function getDuration()
+    public function getDuration(): int
     {
         if (!$this->getCompleted() || !$this->getStarted()) {
             return 0;
         }
-        $duration = $this->getCompleted() - $this->getStarted();
+        $diff = $this->getCompleted()->diff($this->getStarted());
+
+        // convert diff to duration seconds
+        $duration = ($diff->days * 3600 * 24) + ($diff->h * 3600) + ($diff->i * 60) + $diff->s;
```

Comment:
How come we had to change this? it kind of looks like it would of worked beforehand

Reply:
Because I changed `getCompleted` and `getStarted` to be a datetime instead of some random integers.

Reply:
When we say random integers we mean epochs right? but sure ok. I still not fully understand this snippet of code. I think it could be just be (will include in my PR)

```php
$completed = new DateTime();
$started = new DateTime('1 hour ago');

$diff = $completed->getTimestamp() - $started->getTimestamp();

var_dump($diff);
```